app/controllers/lobbytrack_reports_controller.rb

```diff
@@ -60,15 +72,19 @@ def checkin_dates
     date2 = params[:lobbytrack_report][:checkin_date2]
     @dates = "#{date1} to #{date2}"
 
-    begin
-      @lobbytrack_reports = LobbytrackReport.query_checkin_dates(date1, date2)
-    rescue TinyTds::Error => e
-      flash[:error] = e
+    @lobbytrack_form = LobbytrackReport.new(checkin_date1: date1, checkin_date2: date2)
+    if @lobbytrack_form.valid?
+      begin
+        @lobbytrack_reports = LobbytrackReport.query_checkin_dates(date1, date2)
+      rescue TinyTds::Error => e
+        flash[:error] = e
+      end
     end
 
     return if @lobbytrack_reports&.any?
 
     flash[:warning] = "There is no attendance history between the dates #{@dates}"
+    flash[:error] = 'Invalid date entered.'
```

Comment:
This message appears when I put in dates with the datepicker. I would expect to get either the "no attendance history" or "cannot connect" messages.